### PR TITLE
Enhance Number Theory Section with Additional Topics ✨

### LINKED ISSUE #1877: [Idea]: Enhance Number Theory Section with Additional Topics
### Idea Title

Enhance Number Theory Section with Additional Topics

### Idea Description

Currently, the Number Theory section is limited in scope. To provide a more comprehensive overview, please add detailed content for the following additional topics under the Number Theory section:

- Divisibility and Prime Numbers
- Modular Arithmetic 

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by AnushkaChouhan25**
### Description
Added: 
- Divisibility and Prime Numbers
- Modular Arithmetic

Fixes #1877

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Number theory/Divisibility and Prime Numbers.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+id: Divisibility-and-Prime-Numbers
```
Comment (ajay-dhangar) [inline docs/Number theory/Divisibility and Prime Numbers.md]: replace `id: Divisibility-and-Prime-Numbers` with `id: divisibility-and-prime-numbers`

File: docs/Number theory/Modular Arithmetic.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
+---
+id: Modular Arithmetic
```
Comment (ajay-dhangar) [inline docs/Number theory/Modular Arithmetic.md]: replace `id: Modular Arithmetic` with `id: modular-arithmetic`

File: docs/Number theory/Divisibility and Prime Numbers.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+### Prime-Related Concepts
+
+- **Co-primes**: Two numbers are co-prime if their greatest common divisor (GCD) is 1.
+- **Prime Factorization**: Breaking down a number into the product of prime numbers. For example, 60 = 2 * 2 * 3 * 5.
+
+### Prime Number Theorems
+
+1. **Prime Density**: The density of prime numbers decreases as numbers get larger.
+2. **Approximation**: The Prime Number Theorem approximates the number of primes less than a given number \( n \) as \( n / \ln(n) \).
```
Comment (ajay-dhangar) [inline docs/Number theory/Divisibility and Prime Numbers.md]: replace `\( n \) as \( n / \ln(n) \)` with `$n$ as $n \ln(n) $`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by AnushkaChouhan25** [comment]
@ajay-dhangar Done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**